pkg/validations/cluster.go

```diff
@@ -215,3 +216,19 @@ func ValidateK8s128Support(clusterSpec *cluster.Spec) error {
 	}
 	return nil
 }
+
+// ValidateTemplateNameWithKubernetesVersion check if the template name in the machineconfig field includes kubernetes version.
+func ValidateTemplateNameWithKubernetesVersion(spec cluster.Spec) error {
+	replacer := strings.NewReplacer("-", "", ".", "")
+	for _, machineconfig := range spec.CloudStackMachineConfigs {
+		// Replace 1.23, 1-23 to 123 in the template name string.
+		templateName := replacer.Replace(machineconfig.Spec.Template.Name)
+		// Replace 1-23 to 123 in the kubernetesversion string.
+		kubernetesVersion := replacer.Replace(string(spec.Cluster.Spec.KubernetesVersion))
```

Comment:
why do we need to replaces in both strings? isn't it enough with replacing in one?
Also, if we need two, shouldn't it more efficient to have two different replacers?